extra/migrations/20150611000000_migrate_dns_settings.rb

```diff
@@ -0,0 +1,73 @@
+require 'yaml'
+
+class MigrateDnsSettings < ::Proxy::Migration
+  def migrate
+    dns_config = path(src_dir, "settings.d", "dns.yml")
+    if !File.exist?(dns_config)
+      duplicate_original_configuration
+      return
+    end
+
+    to_migrate = YAML.load_file(dns_config)
+    output, unknown = migrate_dns_configuration(to_migrate)
+    copy_original_configuration_except(path("settings.d", "dns.yml"))
+    write_to_files(output, unknown)
+  end
+
+  def known_dns_options
+    {
+        :enabled            => [:dns],
+        :dns_provider       => [:dns],
+        :dns_key            => [:dns_nsupdate, :dns_nsupdate_gss],
+        :dns_server         => [:dns_nsupdate, :dns_nsupdate_gss, :dns_dnscmd],
+        :dns_ttl            => [:dns_nsupdate, :dns_nsupdate_gss],
```

Comment:
dns_ttl is in the main dns.yml file, not the per-provider examples.  The dnscmd provider also tries to access it from dns.yml.